hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,303 @@
+            work_template = publisher.engine.get_template_by_name(
+                work_template_setting.value)
+
+            # store the template on the item for use by publish plugins. we
+            # can't evaluate the fields here because there's no guarantee the
+            # current session path won't change once the item has been created.
+            # the attached publish plugins will need to resolve the fields at
+            # execution time.
+            session_item.properties["work_file_template"] = work_template
+            self.logger.debug(
+                "Work file template defined for Houdini collection.")
+
+        self.logger.info("Collected current Houdini session")
+
+        return session_item
+
+    def collect_node_outputs(self, parent_item):
+        """
+        Creates items for known output nodes
+
+        :param parent_item: Parent Item instance
+        """
+
+        for node_category in _HOUDINI_OUTPUTS:
+            for node_type in _HOUDINI_OUTPUTS[node_category]:
+
+                if node_type == "alembic" and self._alembic_nodes_collected:
+                    self.logger.debug(
+                        "Skipping regular alembic node collection since tk "
+                        "alembic nodes were collected. "
+                    )
+                    continue
+
+                if node_type == "ifd" and self._mantra_nodes_collected:
+                    self.logger.debug(
+                        "Skipping regular mantra node collection since tk "
+                        "mantra nodes were collected. "
+                    )
+                    continue
+
+                path_parm_name = _HOUDINI_OUTPUTS[node_category][node_type]
+
+                # get all the nodes for the category and type
+                nodes = hou.nodeType(node_category, node_type).instances()
+
+                for node in nodes:
+
+                    # get the evaluated path parm value
+                    path = node.parm(path_parm_name).eval()
+
+                    # ensure the output path exists
+                    if not os.path.exists(path):
+                        continue
+
+                    self.logger.info(
+                        "Processing %s node: %s" % (node_type, node.path()))
+
+                    # allow the base class to collect and create the item. it
+                    # should know how to handle the output path
+                    item = super(HoudiniSessionCollector, self)._collect_file(
+                        parent_item,
+                        path,
+                        frame_sequence=True
+                    )
+
+                    # the item has been created. update the display name to
+                    # include the node path to make it clear to the user how it
+                    # was collected within the current session.
+                    item.name = "%s (%s)" % (item.name, node.path())
+
+    def collect_tk_alembicnodes(self, parent_item):
```

Comment:
New collection method for tk alembic nodes. Relies on changes to the app to make this simpler.

Reply:
docstrings pls :)